### PR TITLE
[matmul] change (param-)appl-matmul to use DMA instead of memcpy

### PR DESCRIPTION
**Pull request by yodada**
Replaced memcpy with DMA for evaluation.

### CODE REVIEW

File: examples/cuda/appl-matmul/main.c [hunk 1]
```diff
@@ -188,13 +188,19 @@ int kernel_appl_matmul (int argc, char **argv) {
...
                  * Copy A & B from host onto device DRAM.
                  ******************************************************************************************************************/
-                void *dst = (void *) ((intptr_t) A_device);
-                void *src = (void *) &A[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy A to the device  */
-
-                dst = (void *) ((intptr_t) B_device);
-                src = (void *) &B[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy B to the device */
+                hb_mc_dma_htod_t htod_A = {
```
Comment (mrutt92) [inline examples/cuda/appl-matmul/main.c]: While this is 'correct' (and much better than using memcpy) you'll save significant simulation time by making a 2-length array of `hb_mc_dma_htod_t` and then having a single call to `hb_mc_device_dma_to_device`.

File: examples/cuda/param-appl-matmul/main.c [hunk 2]
```diff
@@ -188,13 +188,19 @@ int kernel_appl_matmul (int argc, char **argv) {
...
                  * Copy A & B from host onto device DRAM.
                  ******************************************************************************************************************/
-                void *dst = (void *) ((intptr_t) A_device);
-                void *src = (void *) &A[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy A to the device  */
-
-                dst = (void *) ((intptr_t) B_device);
-                src = (void *) &B[0];
-                BSG_CUDA_CALL(hb_mc_device_memcpy (&device, dst, src, N * N * sizeof(REAL), HB_MC_MEMCPY_TO_DEVICE)); /* Copy B to the device */
+                hb_mc_dma_htod_t htod_A = {
```
Comment (mrutt92) [inline examples/cuda/param-appl-matmul/main.c]: See above.

**Review by mrutt92 — APPROVED** [review]
Please combine calls to dma_to_device, but then LGTM!

File: examples/cuda/appl-matmul/main.c (hunk 1 above)
Comment (yodada) [reply examples/cuda/appl-matmul/main.c]: Fixed!

**Pull request merged**